h2o-extensions/mojo-pipeline/src/main/java/hex/mojopipeline/MojoPipelineExtension.java

```diff
@@ -0,0 +1,46 @@
+package hex.mojopipeline;
+
+import ai.h2o.mojos.runtime.MojoPipelineFactory;
+import water.AbstractH2OExtension;
+import water.util.Log;
+
+import java.util.ServiceLoader;
+
+public class MojoPipelineExtension extends AbstractH2OExtension {
+
+  private static String NAME = "MojoPipeline";
+
+  private boolean _initialized;
+  private boolean _enabled;
+
+  @Override
+  public String getExtensionName() {
+    return NAME;
+  }
+
+  @Override
+  public void onLocalNodeStarted() {
+  }
+
+  @Override
+  public boolean isEnabled() {
+    if (! _initialized) {
+      initialize();
+    }
+    return _enabled;
+  }
+
+  private synchronized void initialize() {
+    _enabled = hasMojoRuntime();
+    _initialized = true;
+    if (! _enabled) {
+      Log.debug("MOJO Runtime not found");
+    }
+  }
+
+  private boolean hasMojoRuntime() {
+    // relying on implementation - need to improve MOJO2 API
+    return ServiceLoader.load(MojoPipelineFactory.class).iterator().hasNext();
```

Comment:
Maybe safer would be check we have exactly one implementation? To avoid class collisions

Reply:
I think this is okay, it will fail with a better message when people actually try to use the MOJO. This is just a quick check if we have the implementation at all.